xarray/core/dataarray.py

```diff
@@ -110,6 +111,31 @@ def __setitem__(self, key, value):
         self.data_array[pos_indexers] = value
 
 
+class DataArraySizes(collections.Mapping, formatting.ReprMixin):
```

Comment:
Does this need a separate class that holds the array? 
Could it be a `dict` of dim-name to dim-size?